docs/en/styling-and-theming/supplemental.md

```diff
@@ -0,0 +1,642 @@
+# Styling & Theming in Dojo
+
+Dojo widgets function best as simple components that each handle a single responsibility. They should be as encapsulated and modular as possible to promote reusability while avoiding conflicts with other widgets the application may also be using.
+
+Widgets can be styled via regular CSS, but to support encapsulation and reuse goals, each widget should maintain its own independent CSS module that lives parallel to the widget's source code. This allows widgets to be styled independently, without clashing on similar class names used elsewhere in an application.
+
+Dojo differentiates between two types of styling, each representing a different granularity of styling concerns within an enterprise web application:
+
+-   [Structural styles](#structural-widget-styling) (_granularity:_ per-widget)
+    -   The minimum styles necessary for a widget to function. These are localized to the particular widget they are needed for.
+-   [Presentational styles](#theming-a-dojo-application) (_granularity:_ application-wide)
```

Comment:
Is there actually a third category in that case? 'Themeable styles'?

Reply:
Themeable styles are still per widget and not generally shared across different widget

Reply:
@tomdye Can probably clarify this better the me (or tell me I'm wrong of course)

Reply:
it's upto the widget dev which css they pass via `this.theme` and which they do not. The classes not passed to `this.theme` would be things like laying out children etc. 90% of classes should be passed via `this.theme`. The third category might be shared styles that get imported to multiple widgets but this is more of an APP concern than a widget library concern. An example of this could be a typography.m.css file that provides `bold` / `header` classes to be re-used across an app.